urdfenvs/urdfCommon/generic_robot.py

```diff
@@ -65,18 +62,24 @@ def read_limits(self) -> None:
     def set_acceleration_limits(self):
         pass
 
-    def get_indexed_joint_info(self) -> dict:
+    def get_indexed_joint_info(self) -> None:
         """Get indexed joint info.
 
         This function can be used for debugging and finding
         the correct joint indices for self.setJointIndices.
 
         """
-        indexed_joint_info = {}
+        self._robot_joints = [] 
+        self._castor_joints = [] 
         for i in range(p.getNumJoints(self._robot)):
             joint_info = p.getJointInfo(self._robot, i)
-            indexed_joint_info[joint_info[0]] = joint_info[1]
-        return indexed_joint_info
+            joint_name = joint_info[1].decode("UTF-8")
+            print(joint_name)
+            if joint_name in self._joint_names:
+                self._robot_joints.append(i)
+            if "castor" in joint_name:
+                self._castor_joints.append(i)
+        
```

Comment:
Concerning the other comment on repeated code lines: This is where I expect self._urdf_joints to be set, just like you do it with the other joint indices.